src/Hint/Bracket.hs

```diff
@@ -120,6 +120,9 @@ foo' :: ((A) => ()) -> ()
 bar' :: ((A a) => ()) -> ()
 data Dict c where Dict :: (c) => Dict c
 data Dict' c a where Dict' :: (c a) => Dict' c a
+
+issue1501: Redundant bracket hint resulted in a parse error
+\(Proxy @a) -> True
```

Comment:
You may need a binding, as in `x = \(Proxy @a) -> True` - I'm not sure.

Reply:
Actually this would cause HLint to suggest `x (Proxy @a) = True`. You could use `x = f $ \(Proxy @a) -> True`.

Reply:
Do you mean as an extra case or instead of the one already there? :)

Reply:
I meant instead of. This is not needed given the one below.